sway/input/cursor.c

```diff
@@ -791,6 +791,11 @@ static void check_constraint_region(struct sway_cursor *cursor) {
 				wlr_cursor_warp_closest(cursor->cursor, NULL,
 					sx + con->content_x - view->geometry.x,
 					sy + con->content_y - view->geometry.y);
+
+				// The specification says we should send pointer enter/motion events as
+				// necessary, but no relative motion events.
+				seatop_pointer_motion(cursor->seat, get_current_time_msec(), 0, 0);
```

Comment:
We generally rebase the cursor in this case. Would that work here?

Reply:
Almost, but I think there's a very corner case that that wouldn't handle. Cursor rebase only sends a pointer enter event:

https://github.com/swaywm/sway/blob/5d13f647f9384e59012c0f829651911564bb5365/sway/input/seatop_default.c#L656-L659

But, after https://github.com/swaywm/sway/commit/6f0a0bd3852c6d92f30d43b9a39dc743987584ac, we may also need to warp when the region changes on the same surface, in which case `wlr_seat_pointer_notify_enter` will be a no-op (if the surface already has pointer focus), but we probably do want to send a motion event to the client anyway.

Reply:
Should rebasing the cursor also send a motion event? I'd say yes, because rebasing may update the cursor position without changing the focus.

Reply:
Actually, that sounds like a good idea, done. And then in the case where it doesn't have focus already, it behaves as it has until now, since wlroots won't send a motion event after a real enter.

Reply:
Turns out this was a very good call, since `swaymsg seat * cursor move dx dy` didn't notify the underlying surface of the motion (https://github.com/swaywm/sway/issues/5361), and this fixes that too.